packages/frontend-core/src/components/FilterBuilder.svelte

```diff
@@ -67,6 +67,12 @@
 
   const removeFilter = id => {
     filters = filters.filter(field => field.id !== id)
+
+    // Clear all filters when no fields are specified
+    let [first] = filters
+    if (filters.length == 1 && first?.onEmptyFilter) {
```

Comment:
Super NAB but the logic here can be a little simpler, since the first check is that there is only 1 filter, so you don't need to dereference the array or use optional chaining here - it could just be
```js
if (filters.length === 1 && filters[0].onEmptyFilter) {
...
```